src/Sfa.Tl.Matching.Application/Services/ReferralService.cs

```diff
@@ -1,136 +1,123 @@
 ﻿using System.Collections.Generic;
 using System.Linq;
-using System.Text;
 using System.Threading.Tasks;
+using AutoMapper;
 using Sfa.Tl.Matching.Application.Interfaces;
 using Sfa.Tl.Matching.Data.Interfaces;
 using Sfa.Tl.Matching.Domain.Models;
-using Sfa.Tl.Matching.Models.Configuration;
 using Sfa.Tl.Matching.Models.Dto;
 using Sfa.Tl.Matching.Models.Enums;
 
 namespace Sfa.Tl.Matching.Application.Services
 {
     public class ReferralService : IReferralService
     {
-        private readonly MatchingConfiguration _configuration;
-        private readonly IEmailService _emailService;
-        private readonly IEmailHistoryService _emailHistoryService;
-        private readonly IRepository<Opportunity> _opportunityRepository;
+        private readonly IMapper _mapper;
+        private readonly IMessageQueueService _messageQueueService;
+        private readonly IRepository<OpportunityItem> _opportunityItemRepository;
+        private readonly IRepository<BackgroundProcessHistory> _backgroundProcessHistoryRepository;
 
         public ReferralService(
-            MatchingConfiguration configuration,
-            IEmailService emailService,
-            IEmailHistoryService emailHistoryService,
-            IRepository<Opportunity> opportunityRepository)
+                            IMapper mapper,
+                            IMessageQueueService messageQueueService,
+                            IRepository<OpportunityItem> opportunityItemRepository,
+                            IRepository<BackgroundProcessHistory> backgroundProcessHistoryRepository)
         {
-            _configuration = configuration;
-            _emailService = emailService;
-            _emailHistoryService = emailHistoryService;
-            _opportunityRepository = opportunityRepository;
+            _mapper = mapper;
+            _messageQueueService = messageQueueService;
+            _opportunityItemRepository = opportunityItemRepository;
+            _backgroundProcessHistoryRepository = backgroundProcessHistoryRepository;
         }
 
-        public async Task SendEmployerReferralEmail(int opportunityId)
+
+        public async Task ConfirmOpportunities(int opportunityId, string username)
         {
-            var employerReferral = await GetEmployerReferrals(opportunityId);
-            var sb = new StringBuilder();
+            await CompleteSelectedReferrals(opportunityId, username);
+            await CompleteRemainingItems(opportunityId);
```

Comment:
Move both of this call into function side so that you can set it knowingly after the function has successfully sent the emails

Reply:
This is done